src/api/createDeviceMessageHandlingSystem/src/v1/steps.py

```diff
@@ -0,0 +1,125 @@
+from http import HTTPStatus
+
+from domain.api.common_steps.general import parse_event_body
+from domain.api.common_steps.read_product import (
+    parse_path_params,
+    read_product,
+    read_product_team,
+)
+from domain.core.cpm_product.v1 import CpmProduct
+from domain.core.device.v3 import Device
+from domain.core.error import InvalidSpineMhsResponse, NotEprProductError
+from domain.core.product_key.v1 import ProductKeyType
+from domain.core.questionnaire.v3 import Questionnaire, QuestionnaireResponse
+from domain.repository.device_repository.v3 import DeviceRepository
+from domain.repository.questionnaire_repository.v2 import QuestionnaireRepository
+from domain.repository.questionnaire_repository.v2.questionnaires import (
+    QuestionnaireInstance,
+)
+from domain.request_models.v1 import CreateMhsDeviceIncomingParams
+from domain.response.validation_errors import mark_validation_errors_as_inbound
+
+
+@mark_validation_errors_as_inbound
+def parse_mhs_device_payload(data, cache) -> Device:
+    payload: dict = data[parse_event_body]
+    return CreateMhsDeviceIncomingParams(**payload)
+
+
+def get_party_key(data, cache) -> str:
+    product: CpmProduct = data[read_product]
+    party_keys = [
+        key.key_value
+        for key in product.keys
+        if key.key_type is ProductKeyType.PARTY_KEY
+    ]
+    try:
+        (party_key,) = party_keys
+    except ValueError:
+        raise NotEprProductError(
+            "Not an EPR Product: Cannot create MHS device for product without exactly one Party Key"
+        )
+    return party_key
+
+
+def read_spine_mhs_questionnaire(data, cache) -> Questionnaire:
+    return QuestionnaireRepository().read(QuestionnaireInstance.SPINE_MHS)
+
+
+def validate_spine_mhs_questionnaire_response(data, cache) -> QuestionnaireResponse:
+    spine_mhs_questionnaire: Questionnaire = data[read_spine_mhs_questionnaire]
+    payload: CreateMhsDeviceIncomingParams = data[parse_mhs_device_payload]
+    questionnaire_responses = payload.questionnaire_responses
+
+    # Ensure there's a questionnaire named 'spine_mhs' in the responses
+    if "spine_mhs" not in questionnaire_responses:
```

Comment:
there are some predefined enums for the names of the questionnaires to help any future refactor (`QuestionnaireInstance.SPINE_MHS`)